libtenzir/include/tenzir/plugin.hpp

```diff
@@ -1056,21 +1056,23 @@ extern const char* TENZIR_PLUGIN_VERSION;
       ::tenzir::plugin_ptr::make_builtin(__VA_ARGS__, {})
 #  endif
 
-#  define TENZIR_REGISTER_PLUGIN(name)                                         \
-    template <class>                                                           \
+#  define TENZIR_REGISTER_PLUGIN(name, ...)                                    \
+    template <auto>                                                            \
     struct auto_register_plugin;                                               \
     template <>                                                                \
-    struct auto_register_plugin<name> {                                        \
+    struct auto_register_plugin<[] {}> {                                       \
       auto_register_plugin() {                                                 \
         static_cast<void>(flag);                                               \
       }                                                                        \
       static auto init() -> bool {                                             \
+        /* NOLINTBEGIN(cppcoreguidelines-owning-memory) */                     \
         ::tenzir::plugins::get_mutable().push_back(TENZIR_MAKE_PLUGIN(         \
-          new (name), /* NOLINT(cppcoreguidelines-owning-memory) */            \
+          new (name)(__VA_ARGS__),                                             \
```

Comment:
I think you can enable `TENZIR_REGISTER_PLUGIN(tenzir::plugins::quantile::plugin{"median", 0.5})` if you make this `TENZIR_REGISTER_PLUGIN(...)` and `new __VA_ARGS__`.